pandas/core/generic.py

```diff
@@ -6812,8 +6807,6 @@ def fillna(
         inplace = validate_bool_kwarg(inplace, "inplace")
         value, method = validate_fillna_kwargs(value, method)
 
-        self._consolidate_inplace()
```

Comment:
Maybe it was more performant to consolidate a fragmented frame before fillna or replace?

Nonetheless, I agree with the notion of "avoiding side effects" when calling a method

Reply:
Any performance differences in the methods where consolidation no longer happens?

Reply:
i think the main one is going to be in DataFrame.values where subsequent calls could be faster in the status quo if it can be fully consolidated.